dot/network/notifications.go

```diff
@@ -226,6 +179,58 @@ func (s *Service) createNotificationsMessageHandler(
 	}
 }
 
+func (s *Service) handleHandshake(info *notificationsProtocol, stream network.Stream,
+	hs Handshake, peer peer.ID) error {
+	logger.Tracef("received handshake on notifications sub-protocol %s from peer %s, message is: %s",
+		info.protocolID, stream.Conn().RemotePeer(), hs)
+
+	// if we are the receiver and haven't received the handshake already, validate it
+	// note: if this function is being called, it's being called via SetStreamHandler,
+	// ie it is an inbound stream and we only send the handshake over it.
+	// we do not send any other data over this stream, we would need to open a new outbound stream.
+	hsData := info.peersData.getInboundHandshakeData(peer)
+	if hsData != nil {
+		return fmt.Errorf("%w: %s", errInboundHanshakeExists, peer)
+	}
+
+	logger.Tracef("receiver: validating handshake using protocol %s", info.protocolID)
+
+	hsData = newHandshakeData(true, false, stream)
+	info.peersData.setInboundHandshakeData(peer, hsData)
+
+	err := info.handshakeValidator(peer, hs)
+	if err != nil {
+		logger.Tracef(
+			"failed to validate handshake from peer %s using protocol %s: %s",
+			peer, info.protocolID, err)
+		return errCannotValidateHandshake
+	}
+
+	hsData.validated = true
+	info.peersData.setInboundHandshakeData(peer, hsData)
+
+	// once validated, send back a handshake
+	resp, err := info.getHandshake()
+	if err != nil {
+		logger.Warnf("failed to get handshake using protocol %s: %s", info.protocolID, err)
```

Comment:
nit I know it's copy pasta, but since it shows as deltas, one more suggestion...

Why is this logged at the warn level whereas others are logged at the trace level? Would it feel appropriate to remove this log and rely on the caller Trace log instead?



```suggestion
```

Reply:
> Would it feel appropriate to remove this log and rely on the caller Trace log instead?

I prefer not to ignore errors. If we are not returning the err, at least we should log it as warn.